langchain4j-core/src/main/java/dev/langchain4j/data/document/Document.java

```diff
@@ -2,74 +2,43 @@
 
 import dev.langchain4j.data.segment.TextSegment;
 
-import java.util.Objects;
-
 import static dev.langchain4j.internal.Utils.quoted;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
 
 /**
  * Represents an unstructured piece of text that usually corresponds to a content of a single file.
  * This text could originate from various sources such as a text file, PDF, DOCX, or a web page (HTML).
- * Each document may have associated metadata including its source, owner, creation date, etc.
+ * Each document may have associated {@link Metadata} including its source, owner, creation date, etc.
  */
-public class Document {
+public interface Document {
 
     /**
      * Common metadata key for the name of the file from which the document was loaded.
      */
-    public static final String FILE_NAME = "file_name";
+    String FILE_NAME = "file_name";
     /**
      * Common metadata key for the absolute path of the directory from which the document was loaded.
      */
-    public static final String ABSOLUTE_DIRECTORY_PATH = "absolute_directory_path";
+    String ABSOLUTE_DIRECTORY_PATH = "absolute_directory_path";
     /**
      * Common metadata key for the URL from which the document was loaded.
      */
-    public static final String URL = "url";
-
-    private final String text;
-    private final Metadata metadata;
-
-    /**
-     * Creates a new Document from the given text.
-     *
-     * <p>The created document will have empty metadata.
-     *
-     * @param text the text of the document.
-     */
-    public Document(String text) {
```

Comment:
This is a breaking change BTW

Reply:
If [factory methods](https://github.com/langchain4j/langchain4j/pull/1988/files/f5e81855329dce2c83d00a4acaefaa469246ac81#diff-46b454c8594e76873a66fb2865ff7f3a762687e01aec333eeb40b989ef6ec86eR74) will be used, then it should not be "very" breaking. All document loaders are left intact and still work.

Reply:
OK, I guess this should not affect a lot of users, and we are still in 0.X versions, so this breaking change is acceptable